pkg/database/db.go

```diff
@@ -566,19 +649,19 @@ func (db *DB) YieldAll(ctx context.Context, factoryFunc contracts.EntityFactoryF
 // concurrency is controlled via Options.MaxConnectionsPerTable.
 // Entities for which the query ran successfully will be passed to onSuccess.
 func (db *DB) CreateStreamed(
-	ctx context.Context, entities <-chan contracts.Entity, onSuccess ...OnSuccess[contracts.Entity],
+	ctx context.Context, entities <-chan Entity, onSuccess ...OnSuccess[Entity],
 ) error {
 	first, forward, err := com.CopyFirst(ctx, entities)
-	if first == nil {
+	if err != nil {
 		return errors.Wrap(err, "can't copy first entity")
 	}
```

Comment:
Changing the error handling in this way sounds like a good idea (given that the function returns `error` already), but then the following case within the function should probably be changed to also return an error in this case:

https://github.com/Icinga/icingadb/blob/5d182aad64891e7007b7d90320bccce0c933d437/pkg/com/com.go#L52-L54